components/kyma-environment-broker/internal/process/provisioning/inject_btp_operator_credentials_step.go

```diff
@@ -129,14 +129,20 @@ func updateSecretData(secret *apicorev1.Secret, secretFromParameters *apicorev1.
 
 func (s *InjectBTPOperatorCredentialsStep) createOrRetry(k8sClient client.Client, newSecret *apicorev1.Secret, err error, log logrus.FieldLogger) error {
 	if apierrors.IsNotFound(err) {
+		namespace := &apicorev1.Namespace{ObjectMeta: v1.ObjectMeta{Name: secretNamespace}}
+		err = k8sClient.Create(context.Background(), namespace)
+		if err != nil {
+			log.Warnf("could not create %s namespace: %w", secretNamespace, err)
+		}
+
 		err = k8sClient.Create(context.Background(), newSecret)
```

Comment:
how about returning an error earlier if namespace creation fails and it doesn't exist in the cluster?

Reply:
We will fail on secret creation eventually with a message specifying the reason (missing namespace). I have deliberately simplified the logic.

Reply:
Will do.

Reply:
Done.